Client/mods/deathmatch/logic/luadefs/CLuaResourceDefs.cpp

```diff
@@ -352,28 +353,17 @@ int CLuaResourceDefs::GetResourceGUIElement(lua_State* luaVM)
     return 1;
 }
 
-int CLuaResourceDefs::GetResourceDynamicElementRoot(lua_State* luaVM)
+CClientEntity* CLuaResourceDefs::GetResourceDynamicElementRoot(lua_State* luaVM, std::optional<CResource*> resource)
 {
-    CResource*       pResource = NULL;
-    CScriptArgReader argStream(luaVM);
-    argStream.ReadUserData(pResource);
+    CResource* pResource = resource.value_or(&lua_getownerresource(luaVM));
 
-    if (!argStream.HasErrors())
+    CClientEntity* resourceDynamicRoot = pResource->GetResourceDynamicEntity();
```

Comment:
Are you sure there is no possibility for pResource to be nullptr here?

Reply:
Non-ArgumentParser variant didnt have this check and everything worked fine, so...

Reply:
You can add an `assert` to assure that `pResource` is always valid.

Reply:
> You can add an `assert` to assure that `pResource` is always valid.

Shouldn't an exception be thrown instead of using `assert`?

Reply:
I don't think so. Exceptions are supposed to be used when you're able to catch and process them. In this case an invalid ptr means that yours program in fire and there simply no sense in continuing of the execution.